optuna/integration/_lightgbm_tuner/optimize.py

```diff
@@ -955,9 +967,9 @@ def __init__(
         sample_size: Optional[int] = None,
         study: Optional[optuna.study.Study] = None,
         optuna_callbacks: Optional[List[Callable[[Study, FrozenTrial], None]]] = None,
+        model_dir: Optional[str] = None,
```

Comment:
Could you tell me why `model_dir` is moved here?

Reply:
Sure, this cosmetic change was made so that arguments of `LightGBMTunerCV` is ordered similarly with `LightGBMTuner`. Please kindly let me know if any concern, happy to hear your opinion.

Let me think a bit for the unittest for making directory.

Reply:
> this cosmetic change was made so that arguments of LightGBMTunerCV is ordered similarly with LightGBMTuner

I didn't notice. Thank you for pointing it out and addressing it.

> Please kindly let me know if any concern, happy to hear your opinion.

Meanwhile,  leave as it is.

My take: I found it a little bit too drastic because this change breaks backward compatibility.
However, as the number of arguments of `LightGBMTuner.__init__` is large, so maybe we can assume most users pass arguments with the corresponding keywords.
So, I'd say this change comes with a certain amount of risk which can be negligible.

Reply:
Understood, I will cancel the order change. Thank you for explanation!